e2e/machine_migration_test.go

```diff
@@ -0,0 +1,370 @@
+		case machinev1beta1.MachineAuthorityClusterAPI:
+			By("Return the CAPI Machine generation")
+			capiMachine, err := framework.GetMachine(cl, machineName, framework.CAPINamespace)
+			if err != nil {
+				return 0
+			}
+			return int(capiMachine.Generation)
+		case machinev1beta1.MachineAuthorityMachineAPI:
+			By("Return the MAPI Machine generation")
+			return mapiGeneration
+		default:
+			Fail(fmt.Sprintf("unknown authoritative API type: %v", auth))
+			return 0
+		}
+
+	}, framework.WaitShort, framework.RetryShort).Should(Equal(synchronizedGeneration), "synchronizedGeneration is not %s  machine's generation", auth)
+}
+
+func verifyMAPIMachinePausedCondition(ctx context.Context, cl client.Client, machineName string, auth machinev1beta1.MachineAuthority) {
+	var conditionMatcher types.GomegaMatcher
+
+	switch auth {
+	case machinev1beta1.MachineAuthorityMachineAPI:
+		By("Verify the MAPI Machine is Unpaused")
+		conditionMatcher = SatisfyAll(
+			HaveField("Type", Equal(MAPIPausedCondition)),
+			HaveField("Status", Equal(corev1.ConditionFalse)),
+			HaveField("Reason", Equal("AuthoritativeAPIMachineAPI")),
+			HaveField("Message", ContainSubstring("MachineAPI")),
+		)
+	case machinev1beta1.MachineAuthorityClusterAPI:
+		By("Verify the MAPI Machine is Paused")
+		conditionMatcher = SatisfyAll(
+			HaveField("Type", Equal(MAPIPausedCondition)),
+			HaveField("Status", Equal(corev1.ConditionTrue)),
+			HaveField("Reason", Equal("AuthoritativeAPINotMachineAPI")),
+			HaveField("Message", ContainSubstring("ClusterAPI")),
+		)
+	default:
+		Fail(fmt.Sprintf("unknown authoritative API type: %v", auth))
+	}
+
+	Eventually(func() []machinev1beta1.Condition {
+		mapiMachine, err := mapiframework.GetMachine(cl, machineName)
+		if err != nil {
+			return nil
+		}
+		return mapiMachine.Status.Conditions
+	}, framework.WaitMedium, framework.RetryMedium).Should(
+		ContainElement(conditionMatcher),
+		fmt.Sprintf("Expected paused condition for %s not found", auth),
+	)
+}
+
+func verifyCAPIMachinePausedCondition(cl client.Client, machineName string, auth machinev1beta1.MachineAuthority) {
+	var conditionMatcher types.GomegaMatcher
+
+	switch auth {
+	case machinev1beta1.MachineAuthorityClusterAPI:
+		By("Verify the CAPI Machine is Unpaused")
+		conditionMatcher = SatisfyAll(
+			HaveField("Type", Equal(CAPIPausedCondition)),
+			HaveField("Status", Equal(metav1.ConditionFalse)),
+			HaveField("Reason", Equal("NotPaused")),
+		)
+	case machinev1beta1.MachineAuthorityMachineAPI:
+		By("Verify the CAPI Machine is Paused")
+		conditionMatcher = SatisfyAll(
+			HaveField("Type", Equal(CAPIPausedCondition)),
+			HaveField("Status", Equal(metav1.ConditionTrue)),
+			HaveField("Reason", Equal("Paused")),
+		)
+	default:
+		Fail(fmt.Sprintf("unknown authoritative API type: %v", auth))
+	}
+
+	Eventually(func() []metav1.Condition {
+		capiMachine, err := framework.GetMachine(cl, machineName, framework.CAPINamespace)
+		if err != nil {
+			return nil
+		}
```

Comment:
Change the function signature and return the error here as well, then if it continually fails we will get an output of the error, see [this](https://github.com/openshift/cluster-capi-operator/pull/287#discussion_r2158726756)